### PR TITLE
Reproing Blue Screen Of Death if ongoing soft deletion is cancelled.

### PR DESCRIPTION
**Pull request by vimahali**
Repro steps:
1. Create a server folder and client folder (both are local windows folder)
2. Place 1000 dummy files in server folder.
3. Start the package from the cloudmirror solution.
4. Select server and client folder. 1000 placeholders will be created in client folder
5. Now select all the files in client folder -> soft delete them (moving to recycle bin)
6. Cancel in between. 
7. System crashes with BSOD

image

For successful case (without cancelling) this works fine.

Changes:
1. Added OnNotifyDelete and sent acknowledgment.
2. Bumping up sdk version as per this issue: https://github.com/microsoft/Windows-classic-samples/issues/267

### CODE REVIEW

**Comment by vimahali** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by oldnewthing** [comment]
Thanks. I have asked the feature team to review the PR.

File: Samples/CloudMirror/CloudMirror/FakeCloudProvider.cpp [hunk 1]
```diff
@@ -104,6 +105,34 @@ void CALLBACK FakeCloudProvider::OnCancelFetchData(
...
+    CF_OPERATION_PARAMETERS opParams = { 0 };
+
+    wprintf(L"Acknowledging delete request %s\n", callbackInfo->NormalizedPath);
+
+    opInfo.StructSize = sizeof(opInfo);
+    opInfo.Type = CF_OPERATION_TYPE_ACK_DELETE;
+    opInfo.ConnectionKey = callbackInfo->ConnectionKey;
+    opInfo.TransferKey = callbackInfo->TransferKey;
+    opParams.ParamSize = CF_SIZE_OF_OP_PARAM(AckDelete);
+    opParams.AckDelete.CompletionStatus = 0;
```
Comment (abhijeet-gautam) [inline Samples/CloudMirror/CloudMirror/FakeCloudProvider.cpp:128]: opParams.AckDelete.CompletionStatus = STATUS_SUCCESS;

Comment (abhijeet-gautam) [inline Samples/CloudMirror/CloudMirror/FakeCloudProvider.cpp:128]: opParams.AckDelete.CompletionStatus = STATUS_SUCCESS;

File: Samples/CloudMirror/CloudMirror/FakeCloudProvider.cpp [hunk 2]
```diff
@@ -104,6 +105,34 @@ void CALLBACK FakeCloudProvider::OnCancelFetchData(
...
+    opParams.AckDelete.Flags = CF_OPERATION_ACK_DELETE_FLAG_NONE;
+
+    // no issue with CfExecute during happy path.
```
Comment (abhijeet-gautam) [inline Samples/CloudMirror/CloudMirror/FakeCloudProvider.cpp:131]: Thanks for implementing this callback in the sample. 
These comments aren't desirable in the sample though. Please remove.

**Pull request closed without merging**